src/xss-network-keyvaluesort.hpp

```diff
@@ -1,5 +1,19 @@
-#ifndef AVX512_KEYVALUE_NETWORKS
-#define AVX512_KEYVALUE_NETWORKS
+#ifndef XSS_KEYVALUE_NETWORKS
+#define XSS_KEYVALUE_NETWORKS
+
+#include "avx512-64bit-qsort.hpp"
+#include "avx2-64bit-qsort.hpp"
+
+template <int num_lanes>
+struct index_64bit_vector_type;
+template <>
+struct index_64bit_vector_type<8> {
+    using type = zmm_vector<uint64_t>;
+};
+template <>
+struct index_64bit_vector_type<4> {
+    using type = avx2_vector<uint64_t>;
```

Comment:
I think this should be `ymm_vector` and this might be one of the reason why this patch fails on a 32-bit system. On 32-bit, arrsize_t/size_t is only 4 bytes. You could also avoid using `index_64bit_vector_type` by just using a `std::conditional.`